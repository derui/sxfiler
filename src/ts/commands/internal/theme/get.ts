import { Actions } from "@/modules";
import { CommandLike, CommandState, CommandDescriptor } from "@/commands/type";
import { Dispatcher } from "@/types";
import { actions } from "@/modules/theme";
import { GetRequest } from "@/generated/theme_pb";
import { Theme } from "@/rpc/client-procedures";

const identifier = "internal.theme.select";

export type Payload = undefined;
export type Command = CommandLike<Payload>;

/**
 * command descriptor to resolve
 */
export const descriptor: CommandDescriptor<Payload> = Object.freeze({
  identifier,
  payloadSpec: undefined as any,
});

export const createCommand = (): Command => {
  return {
    identifier,
    async execute(dispatcher: Dispatcher<Actions>, args: CommandState) {
      const res = await args.clientResolver.rpcClient().use(Theme.getTheme)(new GetRequest());

      const theme = res.getTheme();
      if (theme) {
        dispatcher.dispatch(actions.update(theme));
      }
    },
  };
};
