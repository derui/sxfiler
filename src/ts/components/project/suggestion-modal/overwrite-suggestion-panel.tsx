import * as React from "react";
import * as Element from "../../ui/element/element";

const style: ClassNames = require("./overwrite-suggestion-panel.module.scss");

type ClassNames = {
  root: string;
  text: string;
};

export type Props = {
  selected: boolean;
};

export const Component: React.FC<Props> = ({ selected }) => {
  return (
    <Element.Component className={style.root} aria-selected={selected}>
      <p className={style.text}>Overwrite</p>
    </Element.Component>
  );
};
