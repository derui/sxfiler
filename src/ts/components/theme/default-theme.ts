// styled-components.ts
import * as styledComponents from "styled-components";

export interface Theme {
  baseFontSize: string;
  baseBorderRadius: string;
  headerShadow: string;
  boxShadow: string;
  colors: {
    base03: string;
    base02: string;
    base01: string;
    base00: string;
    base0: string;
    base1: string;
    base2: string;
    base3: string;
    yellow: string;
    orange: string;
    red: string;
    magenta: string;
    violet: string;
    blue: string;
    cyan: string;
    green: string;
  };
  spaces: {
    large: string;
    base: string;
    nano: string;
    small: string;
  };
}

export const Theme = {
  baseFontSize: "13.5px",
  baseBorderRadius: "4px",
  headerShadow: "0px 7px 3px rgba(0, 0, 0, 0.2)",
  boxShadow: "4px 4px 7px rgba(0, 0, 0, 0.2), -4px 0px 7px rgba(0, 0, 0, 0.2)",

  colors: {
    /* color schema of solalized-dark (from http://ethanschoonover.com/solarized) */
    base03: "#002b36",
    base02: "#073642",
    base01: "#586e75",
    base00: "#657b83",
    base0: "#839496",
    base1: "#93a1a1",
    base2: "#eee8d5",
    base3: "#fdf6e3",
    yellow: "#b58900",
    orange: "#cb4b16",
    red: "#dc322f",
    magenta: "#d33682",
    violet: "#6c71c4",
    blue: "#268bd2",
    cyan: "#2aa198",
    green: "#859900",
  },
  spaces: {
    large: "16px",
    base: "8px",
    small: "4px",
    nano: "2px",
  },
};

const { default: styled, css, ThemeProvider } = styledComponents as styledComponents.ThemedStyledComponentsModule<
  Theme
>;

export { styled, css, ThemeProvider };
