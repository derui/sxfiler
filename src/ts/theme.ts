import { createContext } from "preact";
import { useContext } from "preact/hooks";

// The context to get theme. The implementation of theme in this project is simple CSS class name for DOM.
export const ThemeContext = createContext<string | undefined>(undefined);

// too simple hook to use theme in container component
export const useTheme = () => {
  return useContext(ThemeContext);
};
