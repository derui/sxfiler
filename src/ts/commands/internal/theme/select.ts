import { Actions } from "@/modules";
import { CommandLike, CommandState, CommandDescriptor } from "@/commands/type";
import { Dispatcher } from "@/types";
import { selectTheme } from "@/modules/theme/selectors";
import { actions } from "@/modules/theme";

const identifier = "internal.theme.select";

export type Payload = { name: string };
export type Command = CommandLike<Payload>;

/**
 * command descriptor to resolve
 */
export const descriptor: CommandDescriptor<Payload> = Object.freeze({
  identifier,
  payloadSpec: undefined as any,
});

export const createCommand = (): Command => {
  return {
    identifier,
    execute(dispatcher: Dispatcher<Actions>, args: CommandState, payload: Payload) {
      const theme = selectTheme(args.state.theme, payload.name);

      if (theme) {
        dispatcher.dispatch(actions.select(name));
      }

      return Promise.resolve();
    },
  };
};
