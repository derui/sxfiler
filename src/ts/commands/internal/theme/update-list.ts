import { Actions } from "@/modules";
import { CommandLike, CommandState, CommandDescriptor } from "@/commands/type";
import { Dispatcher } from "@/types";
import { Theme } from "@/rpc/client-procedures";
import { actions } from "@/modules/theme";
import { ListRequest } from "@/generated/theme_pb";

const identifier = "internal.theme.update-list";

export type Payload = undefined;
export type Command = CommandLike<Payload>;

/**
 * command descriptor to resolve
 */
export const descriptor: CommandDescriptor<Payload> = Object.freeze({
  identifier,
  payloadSpec: undefined as any,
});

export const createCommand = (): Command => {
  return {
    identifier,
    async execute(dispatcher: Dispatcher<Actions>, args: CommandState) {
      const req = new ListRequest();
      const res = await args.clientResolver.rpcClient().use(Theme.list)(req);

      dispatcher.dispatch(actions.updateList(res.getThemesList()));
    },
  };
};
