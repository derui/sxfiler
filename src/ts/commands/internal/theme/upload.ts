import { Actions } from "@/modules";
import { CommandLike, CommandState, CommandDescriptor } from "@/commands/type";
import { Dispatcher } from "@/types";
import { actions } from "@/modules/theme";
import { Theme } from "@/rpc/client-procedures";
import { UpdateRequest, ColorPair } from "@/generated/theme_pb";

const identifier = "internal.theme.upload";

export type Payload = { baseTheme?: string; colors: { [p: string]: string } };
export type Command = CommandLike<Payload>;

/**
 * command descriptor to resolve
 */
export const descriptor: CommandDescriptor<Payload> = Object.freeze({
  identifier,
  payloadSpec: undefined as any,
});

export const createCommand = (): Command => {
  return {
    identifier,
    async execute(dispatcher: Dispatcher<Actions>, args: CommandState, payload: Payload) {
      const req = new UpdateRequest();
      req.setBaseTheme(payload.baseTheme || "");

      Object.entries(payload.colors).forEach(([key, value]) => {
        const pair = new ColorPair();
        pair.setName(key);
        pair.setHexColor(value);
        req.addColorPairs(pair);
      });

      const res = await args.clientResolver.rpcClient().use(Theme.update)(req);
      const theme = res.getTheme();
      if (theme) {
        dispatcher.dispatch(actions.update(theme));
      }
    },
  };
};
