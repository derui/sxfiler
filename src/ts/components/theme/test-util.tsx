import * as React from "react";
import { Theme, ThemeProvider } from "./default-theme";

/**
   wrap element with default theme.
 */
export const wrap = (e: React.ReactElement) => {
  return <ThemeProvider theme={Theme}>{e}</ThemeProvider>;
};
