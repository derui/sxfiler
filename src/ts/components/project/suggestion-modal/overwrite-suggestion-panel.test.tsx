import * as React from "react";
import { wrap } from "@/components/theme/test-util";
import renderer from "react-test-renderer";
import { Component } from "./overwrite-suggestion-panel";

describe("Project", () => {
  describe("Overwrite suggestion", () => {
    it("render with overwrite suggestion", () => {
      const tree = renderer.create(wrap(<Component selected={false} />)).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("change style when selected", () => {
      const tree = renderer.create(wrap(<Component selected={true} />)).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
