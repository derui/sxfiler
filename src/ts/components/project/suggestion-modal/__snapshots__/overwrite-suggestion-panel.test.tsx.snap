// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Overwrite suggestion change style when selected 1`] = `
<div
  aria-selected={true}
  className="sc-bdVaJa hTXxV"
>
  <p>
    Overwrite
  </p>
</div>
`;

exports[`Project Overwrite suggestion render with overwrite suggestion 1`] = `
<div
  aria-selected={false}
  className="sc-bdVaJa hTXxV"
>
  <p>
    Overwrite
  </p>
</div>
`;
